90zfsbootmenu/zfsbootmenu-parse-commandline.sh

```diff
@@ -21,10 +21,34 @@ fi
 
 # Let the command line override our host id.
 # shellcheck disable=SC2034
-spl_hostid=$(getarg spl_hostid=)
+cli_spl_hostid=$( getarg spl.spl_hostid ) || cli_spl_hostid=$( getarg spl_hostid )
+if [ -n "${cli_spl_hostid}" ] ; then
+  # Start empty so only valid hostids are set for future use
+  spl_hostid=
+
+  # Test for decimal
+  if (( 10#${cli_spl_hostid} )) >/dev/null 2>&1 ; then
+    spl_hostid="$( printf "%08x" "${cli_spl_hostid}" )"
+  # Test for hex. Requires 0x, if present, to be stripped
+  # The change to cli_spl_hostid isn't saved outside of the test
+  elif (( 16#${cli_spl_hostid#0x} )) >/dev/null 2>&1 ; then
+    spl_hostid="${cli_spl_hostid#0x}"
+    # printf will strip leading 0s if there are more than 8 hex digits
+    # normalize to a maximum of 8, then run through printf to fill in
+    # if there are fewer than 8 digits.
+    spl_hostid="${spl_hostid:0:8}"
+    spl_hostid="$( printf "%08x" "${cli_spl_hostid}" )"
+  # base 10 / base 16 tests fail on 0
+  elif [ "${cli_spl_hostid#0x}" = "0" ] ; then
```

Comment:
If you use `-eq`, this should handle cases with any number of zeros gracefully, but you probably need to suppress stderr in case cli_spl_hostid is some non-numeric string.